### ISSUE TITLE
Checksum calculation may cause clients to receive timeout exception.

### ISSUE THREAD

**Issue by RootG** [kasirgalabs/Tilki#22]
Currently MD5 sum of a file read from the socket is calculated after reading from socket is completed, which adds delay to server's response time . This situation sometimes causes clients to receive read timeout exception especially if the client's operating system has too small read timeout value and client is trying to send large files.

* Instead of reopening file after writing it, calculate MD5 sum on the fly.

or

* Basically, ensure that client waits for a response by setting timeout value manually. I don't think this is a good solution.
